homeworks/oleksandr.khodakivskyi_Moo0oony/7-array-loops/arrays.js

```diff
@@ -0,0 +1,120 @@
+}
+
+console.log(maxNumberOfArr(arrayOne)); // 44
+console.log(maxNumberOfArr(arrayTwo)); // -1
+console.log(maxNumberOfArr(arrayThree)); // 7
+console.log(maxNumberOfArr(arrayFour)); // 5
+console.log(maxNumberOfArr(arrayFive)); // 5
+
+function minNumberOfArr(array) {
+    let minNumber = array[0];
+    for (let i = 0; i < array.length; i++) {
+        if (array[i] < minNumber) {
+            minNumber = array[i];
+        }
+    }
+    return minNumber;
+}
+
+console.log(minNumberOfArr(arrayOne)); // -12
+console.log(minNumberOfArr(arrayTwo)); // -8
+console.log(minNumberOfArr(arrayThree)); // 1
+console.log(minNumberOfArr(arrayFour)); // -3
+console.log(minNumberOfArr(arrayFive)); // -3
+
+function sumNumbersOfArr(array) {
+    let sum = 0;
+    for (let i = 0; i < array.length; i++) {
+        if (!Number.isNaN(array[i]) && array[i] !== undefined) {
+            sum = array[i] + sum;
+        }
+    }
+    return sum;
+}
+
+console.log(sumNumbersOfArr(arrayOne)); // 32
+console.log(sumNumbersOfArr(arrayTwo)); // -11
+console.log(sumNumbersOfArr(arrayThree)); // 11
+console.log(sumNumbersOfArr(arrayFour)); // 6
+console.log(sumNumbersOfArr(arrayFive)); // 6
+
+function onlyNegativeNumbers(array) {
+    const negativeArray = [];
+    let negativeIndex = 0;
+    for (let i = 0; i < array.length; i++) {
+        if (array[i] < 0) {
+            negativeArray[negativeIndex] = array[i];
+            negativeIndex++;
+        }
+    }
+    return negativeArray;
+}
+
+console.log(onlyNegativeNumbers(arrayOne)); // [-5, -12, -3, -3, -2, -3, -1]
+console.log(onlyNegativeNumbers(arrayTwo)); // [-1, -8, -2]
+console.log(onlyNegativeNumbers(arrayThree)); // []
+console.log(onlyNegativeNumbers(arrayFour)); //  [-3]
+console.log(onlyNegativeNumbers(arrayFive)); // [-3]
+
+function onlyPositiveNumbers(array) {
+    const positiveArray = [];
+    let positiveIndex = 0;
+    for (let i = 0; i < array.length; i++) {
+        if (array[i] > 0) {
+            positiveArray[positiveIndex] = array[i];
+            positiveIndex++;
+        }
+    }
+    return positiveArray;
+}
+
+console.log(onlyPositiveNumbers(arrayOne));// [3, 1, 44, 3, 1, 2, 2, 1, 4]
+console.log(onlyPositiveNumbers(arrayTwo)); // [];
+console.log(onlyPositiveNumbers(arrayThree)); // [1, 7, 3];
+console.log(onlyPositiveNumbers(arrayFour)); // [1, 3, 5];
+console.log(onlyPositiveNumbers(arrayFive)); // [1, 3, 5];
+
+// Additional tasks
+// Custom .map method
+
+function customMap(array, chosenFunction) {
+    const editedArray = [];
+    for (let i = 0; i < array.length; i++) {
+        editedArray[i] = chosenFunction(array[i], i, array);
+    }
+
+    return editedArray;
+}
+
+const exampleFunction = (x) => x * 2;
+
+console.log(customMap(arrayOne, exampleFunction));
+
+// Custom .reduce method
+
+function customReduce(array, chosenFunction, initialValue) {
+    let accumulator = initialValue !== undefined ? initialValue : array[0];
+    for (let i = 0; i < array.length; i++) {
```

Comment:
You have an error without accumulator
![image](https://github.com/MastersAcademy/fe-course-basic-2023/assets/98031768/b490139d-6082-4a23-bf09-f1ab37e36f04)